tests/src/test/scala/com/nvidia/spark/rapids/GpuDeviceManagerSuite.scala

```diff
@@ -83,4 +82,25 @@ class GpuDeviceManagerSuite extends AnyFunSuite with BeforeAndAfter {
       GpuDeviceManager.shutdown()
     }
   }
+
+  test("RMM mode pool enabled with deprecated conf being false") {
+    val rapidsConf = new RapidsConf(Map(
+      "spark.rapids.memory.gpu.pooling.enabled" -> "false",
+      RapidsConf.RMM_POOL.key -> "ARENA"))
+    val mode = GpuDeviceManager.rmmModeFromConf(rapidsConf)
+    val isPoolEnabled = (mode & RmmAllocationMode.ARENA) != RmmAllocationMode.CUDA_DEFAULT
+    assert(isPoolEnabled,
+      s"RMM pool should be enabled when ${RapidsConf.RMM_POOL.key} is not 'NONE'")
+  }
+
+  test("RMM mode pool disabled with deprecated conf being true") {
+    val rapidsConf = new RapidsConf(Map(
+      "spark.rapids.memory.gpu.pooling.enabled" -> "true",
+      RapidsConf.RMM_POOL.key -> "NONE"))
+    val mode = GpuDeviceManager.rmmModeFromConf(rapidsConf)
+    val isPoolDisabled =
+      (mode & ~RmmAllocationMode.CUDA_MANAGED_MEMORY) == RmmAllocationMode.CUDA_DEFAULT
+    assert(isPoolDisabled,
+      s"RMM pool should be disabled when ${RapidsConf.RMM_POOL.key} is 'NONE'")
+  }
```

Comment:
Do we still need these tests? If so, please remove `spark.rapids.memory.gpu.pooling.enabled` from them.

Reply:
no, removed